frontend/src/components/v1/chat/event-content-helpers/get-action-content.ts

```diff
@@ -0,0 +1,201 @@
+import { ActionEvent } from "#/types/v1/core";
+import { getDefaultEventContent, MAX_CONTENT_LENGTH } from "./shared";
+import i18n from "#/i18n";
+import { SecurityRisk } from "#/types/v1/core/base/common";
+
+const getRiskText = (risk: SecurityRisk) => {
+  switch (risk) {
+    case SecurityRisk.LOW:
+      return i18n.t("SECURITY$LOW_RISK");
+    case SecurityRisk.MEDIUM:
+      return i18n.t("SECURITY$MEDIUM_RISK");
+    case SecurityRisk.HIGH:
+      return i18n.t("SECURITY$HIGH_RISK");
+    case SecurityRisk.UNKNOWN:
+    default:
+      return i18n.t("SECURITY$UNKNOWN_RISK");
+  }
+};
+
+const getNoContentActionContent = (): string => "";
+
+// File Editor Actions
+const getFileEditorActionContent = (action: ActionEvent["action"]): string => {
```

Comment:
```suggestion
const getFileEditorActionContent = (
  action: FileEditorAction | StrReplaceEditorAction,
): string => {
```

Let’s try and use the narrowed types wherever we can instead of generic types. This will add IDE and TypeScript support for the following properties. That way we won’t have to check like `!("command" in action && "file_text" in action && "path" in action)`

Please make the changes to all of the related handlers you have here and for observations